### ISSUE TITLE
Proposal: Prefer Number methods to their Global counterparts 

### ISSUE THREAD

**Issue by kirly-af** [eslint/eslint#8894]
**Please describe what the rule should do:**
This rule will enforce the use of the modularized versions of `parseInt`, `parseFloat` and `isNaN`.

**What category of rule is this? (place an "X" next to just one item)**

[ ] Enforces code style
[x] Warns about a potential error
[ ] Suggests an alternate way of doing something
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
isNaN(foo); // warn
parseInt(foo); // warn
parseFloat(foo); // warn
```

``` js
Number.isNaN(foo); // fine
Number.parseInt(foo); // fine
Number.parseFloat(foo); // fine
```

**Why should this rule be included in ESLint (instead of a plugin)?**
From MDN: The purpose of Number.parseInt and Number.parseFloat is "modularization of globals". At the same time Number.NaN is "more robust". This rule should be part of the [Ecmascript 6](http://eslint.org/docs/rules/#ecmascript-6) rules subset.

**Comment by not-an-aardvark** [comment]
Would this be covered by [`no-restricted-globals`](http://eslint.org/docs/rules/no-restricted-globals)?

**Comment by kirly-af** [comment]
Well, sure it could (I was not aware of that rule to be honest). However the lint message won't be explicit enough and since most people probably don't know these functions exist on the Number object, it will let them think "wait... Why can't I use isNaN?".

**Comment by kirly-af** [comment]
Also this rule might be auto "fixable".

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get consensus from the team, so I'm closing it. We define [consensus](http://eslint.org/docs/maintainer-guide/issues.html#consensus) as having three :+1:s from team members, as well as a team member willing to champion the proposal. This is a high bar by design -- we can't realistically accept and maintain every feature request in the long term, so we only accept feature requests which are useful enough that there is consensus among the team that they're worth adding.

Since ESLint is pluggable and can load custom rules at runtime, the lack of consensus among the ESLint team doesn't need to be a blocker for you using this in your project, if you'd find it useful. It just means that you would need to [implement the rule](http://eslint.org/docs/developer-guide/working-with-rules) yourself, rather than using a bundled rule that is packaged with ESLint.

**Comment by platinumazure** [comment]
@kirly-af By the way, [`no-restricted-globals`](https://eslint.org/docs/rules/no-restricted-globals) now supports custom messages, so you could use it and specify whatever message you want. Hope this helps!
